tensorflow/python/compiler/tensorrt/utils.py

```diff
@@ -86,3 +88,97 @@ def is_experimental_feature_activated(feature_name):
   return (feature_name
           in os.environ.get("TF_TRT_EXPERIMENTAL_FEATURES",
                             default="").split(","))
+
+
+def get_nodes_in_engine(graphdef, node_name):
+  ops_in_engine = collections.defaultdict(int)
+  for func in graphdef.library.function:
+    if f"{node_name}_native_segment" == func.signature.name:
+      node_count = len(func.node_def)
+      for node in func.node_def:
+        ops_in_engine[node.op] += 1
+      break
+  return node_count, ops_in_engine
+
+
+def export_graph_to_dot(graphdef, dot_output_filename):
+
+  device_clr = dict()
+
+  with open(dot_output_filename, "w") as f:
+    print("digraph graphname {", file=f)
+
+    print("  graph [fontsize=10 fontname=\"Verdana\"];", file=f)
+    # ColorScheme Documentation: https://graphviz.org/doc/info/colors.html
+    print("  node [style=filled height=0.55 colorscheme=set312 shape=box];",
+          file=f
+    )
+
+    # Parsing the graph and collecting node devices existing in the graph.
+    print("\n  subgraph cluster_legend {", file=f)
+    print("    label=\"Device Legend\";", file=f)
+    print("    margin=\"30\";", file=f)
+    print("    node [width=2];", file=f)
+    # prev_device = None
+    for node in graphdef.node:
+      d_lbl = node.device.split("/")[-1]
+      d_lbl = d_lbl if d_lbl else "device:None"
+      if d_lbl not in device_clr:
+        device_clr[d_lbl] = len(device_clr) + 1
+        d_name = d_lbl.replace(":", "_")
+        print(f"    {d_name} [fillcolor={device_clr[d_lbl]} "
+              f"label=\"{d_lbl}\"];", file=f)
+    print("  }", file=f)
+
+    # Parsing the graph and drawing OPs one by one.
+    print("\n  subgraph tensorflow_graph {", file=f)
+    # print("    rank=\"same\";", file=f)
+    print("    node [width=1.35];", file=f)
+    nodes_with_no_inputs = []
+    for node in graphdef.node:
+      output_name = node.name
+
+      try:
+        device_key = node.device.split("/")[-1]
+        device_clr_id = device_clr[device_key if device_key else "device:None"]
+      except:
+        device_clr_id = "white"
+
+      if node.op == "TRTEngineOp":
+        node_count, _ = get_nodes_in_engine(graphdef, output_name)
+        node_label = f"{output_name} [{node_count}]"
+        print(
+            f"  \"{output_name}\" [label=\"{output_name} [{node_count}]\"];",
+            file=f
+        )
```

Comment:
Shall we allow user to point to the TRTEngine node to inspect the details there, in a follow PR?

Reply:
Yes there are multiple possible improvements we could bring to this graph feature. This is one of them. As for now it's a first iteration of this tool. I expect this tool to grow as we need.